src/renderer/api/kube-object.ts

```diff
@@ -56,7 +56,10 @@ export class KubeStatus {
   public readonly reason: string;
 
   constructor(data: IKubeStatus) {
-    Object.assign(this, data);
```

Comment:
Should we check that `data.kind` is actually `Status` here and if not throw an error?